### PR TITLE
fix(setup): out-of-memory error

### PR DESCRIPTION
**Pull request by avara1986**
When running setup.py extensions with the CMake parameter "-j", it could potentially raise an out-of-memory error. If someone wants to expedite the ddtrace installation, they should manually set the "CMAKE_BUILD_PARALLEL_LEVEL" environment variable.
## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [X] Testing strategy is described if automated tests are not included in the PR.
- [X] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [X] Change is maintainable (easy to change, telemetry, documentation).
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by christophe-papazian — DISMISSED** [review]

**Review by juanjux — DISMISSED** [review]

**Comment by majorgreys** [comment]
cc @brettlangdon

File: setup.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def build_extension(self, ext):
                     build_args += ["-j{}".format(self.parallel)]
                 else:
                     # Let CMake determine the parallelism to use
-                    build_args += ["-j"]
+                    build_args += ["-j1"]
```
[the remark this answers is not in the document]
Comment (avara1986) [reply setup.py:350]: your're right, the comment line is out of date TY!

**Review by juanjux — DISMISSED** [review]

File: setup.py [hunk 2]
```diff
@@ -346,8 +346,7 @@ def build_extension(self, ext):
                 if hasattr(self, "parallel") and self.parallel:
                     build_args += ["-j{}".format(self.parallel)]
                 else:
-                    # Let CMake determine the parallelism to use
-                    build_args += ["-j"]
+                    build_args += ["-j1"]
```
Comment (brettlangdon) [inline setup.py:349]: We can probably just remove this else statement? iirc before I made the change, we didn't add the argument at all.

**Review by brettlangdon — COMMENTED** [review]
Can we also update our CircleCI and GitHub Actions (?) to set `CMAKE_BUILD_PARALLEL_LEVEL`?

This build process has a large impact on the time it takes to build the base venvs, running pre-check tasks, and for releasing the library.

Do we know the cause of the OOM errors? Why does building use so much memory when run in parallel? Is this an issue for internal systems or for customers?

If this is an issue for internal systems only, can we set `CMAKE_BUILD_PARALLEL_LEVEL` there instead of removing the build performance optimization here?

**Comment by avara1986** [comment]
> Can we also update our CircleCI and GitHub Actions (?) to set `CMAKE_BUILD_PARALLEL_LEVEL`?

Ok

> This build process has a large impact on the time it takes to build the base venvs, running pre-check tasks, and for releasing the library.
> 
> Do we know the cause of the OOM errors? Why does building use so much memory when run in parallel? Is this an issue for internal systems or for customers?
Many of our reliability environments raises a c++: fatal error: Killed signal terminated program cc1plus and our dashboards report:
```
Process cc1plus (pid: 4101216) triggered an OOM kill on process cc1plus (pid: 4101198). The process had reached 524288 pages in size.

This OOM kill was invoked by a cgroup, containerID: XXX
```

> If this is an issue for internal systems only, can we set `CMAKE_BUILD_PARALLEL_LEVEL` there instead of removing the build performance optimization here?

**Review by mabdinur — APPROVED** [review]
LGTM. You'll also need a review from someone in the python guild

**Review by juanjux — APPROVED** [review]

**Pull request merged**